From 4cada9a726701aaa03a5d84a02e2d2f5c4052435 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sat, 19 Jun 2021 17:49:31 +0000
Subject: [PATCH] libstdc++: add copy constructor for PendingMessage

libstdc++ requires a copy constructor for creating a reference to
std::queue<PendingMessage> from base::flat_map<Key, Mapped>[].
---
 chrome/browser/push_messaging/push_messaging_service_impl.cc | 1 +
 chrome/browser/push_messaging/push_messaging_service_impl.h  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/chrome/browser/push_messaging/push_messaging_service_impl.cc b/chrome/browser/push_messaging/push_messaging_service_impl.cc
index b80559c..3c1e04b 100644
--- a/chrome/browser/push_messaging/push_messaging_service_impl.cc
+++ b/chrome/browser/push_messaging/push_messaging_service_impl.cc
@@ -191,6 +191,7 @@ PendingMessage::PendingMessage(std::string app_id, gcm::IncomingMessage message)
     : app_id(std::move(app_id)),
       message(std::move(message)),
       received_time(base::Time::Now()) {}
+PendingMessage::PendingMessage(const PendingMessage& other) = default;
 PendingMessage::PendingMessage(PendingMessage&& other) = default;
 PendingMessage& PendingMessage::operator=(PendingMessage&& other) = default;
 PendingMessage::~PendingMessage() = default;
diff --git a/chrome/browser/push_messaging/push_messaging_service_impl.h b/chrome/browser/push_messaging/push_messaging_service_impl.h
index 09be58d..93c18e8 100644
--- a/chrome/browser/push_messaging/push_messaging_service_impl.h
+++ b/chrome/browser/push_messaging/push_messaging_service_impl.h
@@ -68,6 +68,7 @@ class InstanceIDDriver;
 namespace {
 struct PendingMessage {
   PendingMessage(std::string app_id, gcm::IncomingMessage message);
+  PendingMessage(const PendingMessage& other);
   PendingMessage(PendingMessage&& other);
   ~PendingMessage();
 
-- 
2.31.1

